backend/src/datasource.js

```diff
@@ -0,0 +1,67 @@
+import { DataSource } from "apollo-datasource";
+import crypto from "crypto";
+
+export class Post {
+    constructor(data) {
+        this.id = crypto.randomBytes(16).toString("hex");
```

Comment:
When generating random IDs it's (in my experience) "business standard" to reuse a cryptographically sound library to generate UUIDs like [uuidjs/uuid](https://github.com/uuidjs/uuid)